controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -17,48 +17,270 @@ limitations under the License.
+	// RKE2CommonNodeConfig is an inline struct that references common attribtes between agent and server nodes
+	v1alpha1.RKE2CommonNodeConfig `json:",inline"`
+
+	// BindAddress describes the rke2 bind address (default: 0.0.0.0).
+	//+optional
+	BindAddress string `json:"bindAddress,omitempty"`
+
+	// AdvertiseAddress IP address that apiserver uses to advertise to members of the cluster (default: node-external-ip/node-ip).
+	//+optional
+	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+
+	// TLSSan Add additional hostname or IP as a Subject Alternative Name in the TLS cert.
+	//+optional
+	TLSSan []string `json:"tlsSan,omitempty"`
+
+	// ServiceNodePortRange is the port range to reserve for services with NodePort visibility (default: "30000-32767").
+	//+optional
+	ServiceNodePortRange string `json:"service-node-port-range,omitempty"`
+
+	// ClusterDNS is the cluster IP for CoreDNS service. Should be in your service-cidr range (default: 10.43.0.10).
+	//+optional
+	ClusterDNS string `json:"clusterDNS,omitempty"`
+
+	// ClusterDomain is the cluster domain name (default: "cluster.local").
+	//+optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
+
+	// DisableComponents lists Kubernetes components and RKE2 plugin components that will be disabled.
+	//+optional
+	DisableComponents DisableComponents `json:"disableComponents,omitempty"`
+
+	// LoadBalancerPort Local port for supervisor client load-balancer. If the supervisor and apiserver are not colocated an additional port 1 less than this port will also be used for the apiserver client load-balancer (default: 6444).
+	//+optional
+	LoadBalancerPort int `json:"lbServerPort,omitempty"`
+
+	// CNI describes the CNI Plugins to deploy, one of none, calico, canal, cilium; optionally with multus as the first value to enable the multus meta-plugin (default: canal).
+	// +kubebuilder:validation:Enum=none;calico;canal;cilium
+	//+optional
+	CNI CNI `json:"cni,omitempty"`
+
+	// PauseImage Override image to use for pause.
+	//+optional
+	PauseImage string `json:"pauseImage,omitempty"`
+
+	// RuntimeImage Override image to use for runtime binaries (containerd, kubectl, crictl, etc).
+	//+optional
+	RuntimeImage string `json:"runtimeImage,omitempty"`
+
+	// CloudProviderName  Cloud provider name.
+	//+optional
+	CloudProviderName string `json:"cloudProviderName,omitempty"`
+
+	// CloudProviderConfigMap  is a reference to a ConfigMap containing Cloud provider configuration.
+	//+optional
+	CloudProviderConfigMap corev1.ObjectReference `json:"cloudProviderConfigMap,omitempty"`
+
+	// NOTE: this was only profile, changed it to cisProfile.
+
+	// AuditPolicySecret Path to the file that defines the audit policy configuration.
+	//+optional
+	AuditPolicySecret corev1.ObjectReference `json:"auditPolicySecret,omitempty"`
+
+	// ControlPlaneResourceRequests Control Plane resource requests.
+	//+optional
+	ControlPlaneResourceRequests string `json:"controlPlaneResourceRequests,omitempty"`
```

Comment:
The requests and limits would be on the Deployment spec for the providers controller manager. We could probably remove these.

Reply:
Done!